### PR TITLE
Add group

### PR DESCRIPTION
**Pull request by Bathande**

### CODE REVIEW

File: skietbaan_fe/package.json [hunk 1]
```diff
@@ -11,7 +11,9 @@
     "collapsible": "^1.0.1",
     "font-awesome": "^4.7.0",
     "history": "^4.7.2",
+
```
Comment (FredJEhlers) [inline skietbaan_fe/package.json]: no need for the whitespace and line 16

File: skietbaan_fe/src/App.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (FredJEhlers) [inline skietbaan_fe/src/App.js]: remove empty line

File: skietbaan_fe/src/App.js [hunk 3]
```diff
@@ -82,19 +83,23 @@ class App extends Component {
...
-							<Route path="/AddMembersGroup" component={AddMembersGroup} exact />
-							<Route path="/EditGroup" component={EditGroup} exact />
-							<Redirect from="/" to="/home" />
-						</Switch>
-					</Router>
-				</div>
-			</Provider>
-		);
-	}
+                <Route path="/viewmembers" component={ViewMembers} exact />
```
Comment (FredJEhlers) [inline skietbaan_fe/src/App.js]: camelCase path's value please

File: skietbaan_fe/src/App.js [hunk 4]
```diff
@@ -82,19 +83,23 @@ class App extends Component {
...
+                <Route
+                  path="/AddMembersGroup"
```
Comment (FredJEhlers) [inline skietbaan_fe/src/App.js]: camelCase not CapitalCase

File: skietbaan_fe/src/App.js [hunk 5]
```diff
@@ -82,19 +83,23 @@ class App extends Component {
...
+                  component={AddMembersGroup}
+                  exact
+                />
+                <Route path="/EditGroup" component={EditGroup} exact />
```
Comment (FredJEhlers) [inline skietbaan_fe/src/App.js]: camelCase not CapitalCase

File: skietbaan_fe/src/actions/types.js [hunk 6]
```diff
@@ -5,12 +5,12 @@ export const NEW_POST = 'NEW_POST';
 export const NEW_COMP = 'NEW_COMP';
 export const FETCH_COMP = 'FETCH_COMP';
 export const UPDATE_COMP_STATUS = 'UPDATE_COMP_STATUS';
-export const BASE_URL = "https://api.skietbaan.retrotest.co.za";
+// export const BASE_URL = "https://api.skietbaan.retrotest.co.za";
+ export const BASE_URL = "http://localhost:50209";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/types.js]: fix this localhost

File: skietbaan_fe/src/components/AddGroup.js [hunk 7]
```diff
@@ -19,52 +23,83 @@ class AddGroup extends Component {
...
   onClick() {
-    let RequestObject = {
-      Name: this.state.name
-    };
-    this.props.groupActions.getname(RequestObject);
-    history.push("/Groups");
+    if (this.state.groups.indexOf(this.state.name) == -1) {
+      if (this.state.name.length != 0) {
+        this.props.getName(this.state.name);
+        // history.push("/Groups");
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddGroup.js]: GC

File: .gitignore [hunk 8]
```diff
@@ -1 +1,15 @@
 *.orig
+/skietbaan_fe/src/actions/types.js.orig
```
Comment (FredJEhlers) [inline .gitignore]: just delete these the first entry should be fine to cover the `.orig` files
why are you ignoring normal things like `/skietbaan_fe/src/reducers/index.js`

File: skietbaan_fe/src/actions/postActions.js [hunk 9]
```diff
@@ -50,7 +50,7 @@ export const createGroups = usersadded => dispatch => {
   /** the method to post new data from the PostForm */
 }
 export const createPost = users => dispatch => {
-  fetch(BASE_URL+"/api/groups/add", {
+  fetch(BASE_URL + "/api/groups", {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/postActions.js]: same as createGroups. please make sure this is the right end point

File: skietbaan_fe/src/components/AddGroup.js [hunk 10]
```diff
@@ -19,52 +24,85 @@ class AddGroup extends Component {
...
+      
+      } else {
+        this.setState({ txt: "group name can't be empty" });
+      }
+    } else {
+      this.setState({ exist: false });
+    }
+  }
+
+  UNSAFE_componentWillMount() {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddGroup.js]: PLease change this to `unsafeComponentWillMount`

File: skietbaan_fe/src/components/AddGroup.js [hunk 11]
```diff
@@ -19,52 +24,85 @@ class AddGroup extends Component {
...
-            autoComplete="off"
-          />
-          <button
-            className={this.state.name == "" ? "add" : "add2"}
-            type="submit"
-            onClick={this.onClick}
-          >
-            Add Users
-          </button>
+      <div className="addgroup-main">
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddGroup.js]: Please change to fully be dash-notation eg `add-group-main`

File: skietbaan_fe/src/components/AddGroup.js [hunk 12]
```diff
@@ -19,52 +24,85 @@ class AddGroup extends Component {
...
+        <div className="page">
+          <div className="TheNavBar">
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddGroup.js]: dash-notation please for all css classes

File: skietbaan_fe/src/components/AddGroup.js [hunk 13]
```diff
@@ -19,52 +24,85 @@ class AddGroup extends Component {
...
+           <div>
+            <img
+              className="backImage"
+              onClick={this.onBack}
+              src={back}
+              alt=""
+            /></div>
+              <label className="center_label">Create Groups</label>
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddGroup.js]: dash-notation is with a dash - not underscore _

File: skietbaan_fe/src/components/AddMembersGroup.js [hunk 14]
```diff
@@ -2,41 +2,49 @@ import React, { Component } from "react";
...
+                ...users,
+                highlighted: false,
+                backgrnd: "white"
+              };
+            })
+          });
         });
-      });
+    } else {
+      this.props.history.push("/ViewGroups");
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddMembersGroup.js]: camelCase

File: skietbaan_fe/src/components/AddMembersGroup.js [hunk 15]
```diff
@@ -49,7 +57,7 @@ class AddMembersGroup extends Component {
       users: this.state.newArray,
       GroupIds: this.props.id
     };
-    fetch(BASE_URL+"/api/groups/postMember/", {
+    fetch(BASE_URL + "/api/groups/postMember/", {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddMembersGroup.js]: above it was `api/Groups/` so which one is it?

File: skietbaan_fe/src/components/AddMembersGroup.js [hunk 16]
```diff
@@ -63,16 +71,18 @@ class AddMembersGroup extends Component {
   }
   toggleHighlight = (name, event) => {
     if (this.state.posts[event].highlighted === true) {
+      this.state.posts[event].backgrnd = "white";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddMembersGroup.js]: use hex values

File: skietbaan_fe/src/components/AddMembersGroup.js [hunk 17]
```diff
@@ -63,16 +71,18 @@ class AddMembersGroup extends Component {
...
     } else {
       this.setState({ selected: name });
+      this.state.posts[event].backgrnd = "#F3F4F9";
       this.state.posts[event].highlighted = true;
       this.setState({ count: this.state.count + 1 });
     }
   };
   onBack() {
-    history.push("/EditGroup");
+    this.props.history.push("/EditGroup");
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddMembersGroup.js]: camelCase

File: skietbaan_fe/src/components/AddMembersGroup.js [hunk 18]
```diff
@@ -109,23 +124,17 @@ class AddMembersGroup extends Component {
     return (
       <main className="TheMain">
         <div className="TheNavBar">
-          <a href="#" class="fa fa-angle-left" onClick={this.onBack} />
-          <div className="center_label">
-            <b>{this.props.name}</b>
-          </div>
+          <img className="backImage" onClick={this.onBack} src={back} alt="" />
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddMembersGroup.js]: dash-notation

File: skietbaan_fe/src/components/AddMembersGroup.js [hunk 19]
```diff
@@ -137,15 +146,42 @@ class AddMembersGroup extends Component {
...
-          </div>
-          <div className="botname">
-            <button className="updates">Update Group</button>
+        {this.state.count==0  ? null : (
+          <div className="bpanel">
+            <table className="group-delete-table">
+              <tbody>
+                <tr>
+                  <td>
+                    <div className="thetextname">Delete</div>
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddMembersGroup.js]: dash notation

File: skietbaan_fe/src/components/EditGroup.js [hunk 20]
```diff
@@ -1,90 +1,108 @@
...
+                highlighted: true,
+                backgrnd: "#F3F4F9",
+                image: marked
+              };
+            })
+          });
         });
-      });
+    } else {
+      this.props.history.push("/ViewGroups");
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: camelCase

File: skietbaan_fe/src/components/EditGroup.js [hunk 21]
```diff
@@ -1,90 +1,108 @@
...
   }
   onChange(event) {
     this.setState({ filterText: event.target.value });
   }
 
-  handleOnClick() {
-    this.setState({count:0})
+  delete = () => {
+    this.setState({ count: 0 });
+    console.log(this.state.newArray);
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: remove debug log

File: skietbaan_fe/src/components/EditGroup.js [hunk 22]
```diff
@@ -1,90 +1,108 @@
...
     const { newArray } = this.state;
     const newarry = [...this.state.posts];
 
-    newarry.splice(this.state.index, 1);
-    this.setState({ posts: newarry });
     for (var i = 0; i < this.state.posts.length; i++) {
-      if (this.state.posts[i].highlighted === true) {
+      if (this.state.posts[i].highlighted === false) {
+        let indexofs=newArray.indexOf(this.state.posts[i])
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: camelCase

File: skietbaan_fe/src/components/AddMembersGroup.js (hunk 16 above)
Comment (FredJEhlers) [reply skietbaan_fe/src/components/AddMembersGroup.js]: make use of the full word `background`

File: skietbaan_fe/src/components/EditGroup.js [hunk 23]
```diff
@@ -1,90 +1,108 @@
...
+        delete this.state.posts[i].colors;
+        delete this.state.posts[i].backgrnd;
+        delete this.state.posts[i].image;
+        delete this.state.posts[i].highlighted;
+        delete this.state.posts[i].id;
       }
-      delete this.state.posts[i].highlighted;
-      delete this.state.posts[i].id;
     }
+    this.setState({ posts: newarry });
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: newArray

File: skietbaan_fe/src/components/EditGroup.js [hunk 24]
```diff
@@ -1,90 +1,108 @@
...
+
     let request = {
-      GroupIds:this.props.id,
+      GroupIds: this.props.id,
       users: this.state.newArray
     };
-    fetch(BASE_URL+"/api/groups/deleteMember/", {
+    fetch(BASE_URL + "/api/groups/deleteMember/", {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: api/Groups

File: skietbaan_fe/src/components/EditGroup.js [hunk 25]
```diff
@@ -1,90 +1,108 @@
...
       .catch(function(data) {});
-  }
-  toggleHighlight = event => {
+  };
+  toggleHighlight = (user, event) => {
+    this.setState({ selected: user });
     if (this.state.posts[event].highlighted === true) {
       this.state.posts[event].highlighted = false;
+      this.state.posts[event].image = redbox;
+      this.state.posts[event].backgrnd = "white";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: remember to change this to background and use hex

File: skietbaan_fe/src/components/EditGroup.js [hunk 26]
```diff
@@ -1,90 +1,108 @@
...
-
+      this.state.posts[event].image = marked;
+      this.state.posts[event].backgrnd = "#F3F4F9";
       this.setState({ count: this.state.count + 1 });
     }
   };
+  rt;
   onBack() {
-    history.push("/ViewGroups");
+    this.props.history.push("/ViewGroups");
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: camelCase

File: skietbaan_fe/src/components/EditGroup.js [hunk 27]
```diff
@@ -1,90 +1,108 @@
...
   }
 
   goToNext = () => {
-    history.push("/AddMembersGroup");
+    this.props.history.push("/AddMembersGroup");
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: camelCase

File: skietbaan_fe/src/components/EditGroup.js [hunk 28]
```diff
@@ -101,15 +119,28 @@ class EditGroup extends Component {
...
-              <li class="list-group-item list-group-item-light" key={post.id}>
-                <input
-                  type="checkbox"
-                  className="boxs"
-                  onClick={() => this.toggleHighlight(index)}
+              <li
+                class="list-group-item list-group-item-light"
+                key={post.id}
+                style={{
+                  background: post.backgrnd
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: background

File: skietbaan_fe/src/components/EditGroup.js [hunk 29]
```diff
@@ -101,15 +119,28 @@ class EditGroup extends Component {
...
+                  onClick={() => this.toggleHighlight(post.username, index)}
+                  src={post.image}
+                  alt=""
                 />
+
                 <label className="blabe">
-                  <div className="userName"> {post.username}</div>
-                  <div className="emails">{post.email}</div>
+                  <div className="userName" className={post.image==marked?"userName":"userName-active"}>
+                    {" "}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: use styling to create space if you need it

File: skietbaan_fe/src/components/EditGroup.js [hunk 30]
```diff
@@ -119,17 +150,14 @@ class EditGroup extends Component {
     return (
       <main className="TheMain">
         <div className="TheNavBar">
-          <a href="#" class="fa fa-angle-left" onClick={this.onBack} />
-          <div className="center_label">
-            {this.props.name}
-          </div>
+          <img className="backImage" onClick={this.onBack} src={back} alt="" />
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: dash-notation

File: skietbaan_fe/src/components/EditGroup.js [hunk 31]
```diff
@@ -145,13 +173,43 @@ class EditGroup extends Component {
...
-          </button>
-        </label>}
+          </div>
+        {this.state.count==0  ? null : (
+          <div className="bpanel">
+            <table className="group-delete-table">
+              <tbody>
+                <tr>
+                  <td>
+                    <div className="thetextname">Delete</div>
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/EditGroup.js]: dash-notation

File: skietbaan_fe/src/components/Groups.js [hunk 32]
```diff
@@ -1,46 +1,62 @@
...
+                highlighted: false,
+                backgrnd:"white",
+                image:unmarked
+              };
+            })
+          });
         });
-      });
+    } else {
+      this.props.history.push("/AddGroup");
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: camelCase

File: skietbaan_fe/src/components/Groups.js [hunk 33]
```diff
@@ -1,46 +1,62 @@
...
+    }
+    fetch("http://localhost:50209/api/Groups")
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: no Localhost

File: skietbaan_fe/src/components/Groups.js [hunk 34]
```diff
@@ -49,10 +65,8 @@ class Groups extends Component {
       delete this.state.posts[i].highlighted;
       delete this.state.posts[i].id;
     }
-    let request = {
-      newArray: this.state.newArray
-    };
-    fetch(BASE_URL+"/api/groups/add", {
+
+    fetch(BASE_URL + "/api/groups/add", {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: api/Groups

File: skietbaan_fe/src/components/Groups.js [hunk 35]
```diff
@@ -69,33 +83,41 @@ class Groups extends Component {
     if (this.state.check == "Select all") {
       for (var i = 0; i < this.state.posts.length; i++) {
         newarry[i].highlighted = true;
+        this.state.posts[i].backgrnd = "#F3F4F9";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: background

File: skietbaan_fe/src/components/Groups.js [hunk 36]
```diff
@@ -69,33 +83,41 @@ class Groups extends Component {
...
       }
-      this.setState({ check: "Unselect" });
+      this.setState({ check: "Unselect all" });
+      this.setState({ posts: newarry });
     } else {
       for (var i = 0; i < this.state.posts.length; i++) {
         newarry[i].highlighted = false;
+        this.state.posts[i].backgrnd = "white";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: background and use hex

File: skietbaan_fe/src/components/Groups.js [hunk 37]
```diff
@@ -69,33 +83,41 @@ class Groups extends Component {
...
       }
       this.setState({ check: "Select all" });
+      this.setState({ posts: newarry });
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: newArray

File: skietbaan_fe/src/components/AddGroup.js (hunk 10 above)
Comment (Bathande) [reply skietbaan_fe/src/components/AddGroup.js]: it doesn't work if i take it put

File: skietbaan_fe/src/components/AddMembersGroup.js (hunk 15 above)
Comment (Bathande) [reply skietbaan_fe/src/components/AddMembersGroup.js]: different controller

File: skietbaan_fe/src/components/AddGroup.js [hunk 38]
```diff
@@ -19,52 +24,85 @@ class AddGroup extends Component {
...
+            </div>
+        
+          <div className="middle_bar">
```
Comment (GeromeSchutte) [inline skietbaan_fe/src/components/AddGroup.js]: dash-notation

File: skietbaan_fe/src/components/AddMembersGroup.js (hunk 14 above)
Comment (GeromeSchutte) [reply skietbaan_fe/src/components/AddMembersGroup.js]: UNSAFE_componentWillMount is a lifecycle method defined in React.Component :)

**Pull request merged**